src/Main/TodoItem.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable jsx-a11y/no-autofocus */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../utils/types/type';
+
+type ClassName = 'completed' | 'view' | 'editing';
+
+type Props = {
+  item: Todo,
+  setTodos: (todos: Todo[]) => void,
+  todos: Todo[],
+  setIsProcessing: (ids: number[]) => void,
+  isProcessing: number[],
+};
+
+export const TodoItem:React.FC<Props> = ({
+  item, setTodos, todos, isProcessing, setIsProcessing,
+}) => {
+  const { id, title, completed } = item;
+  const [editingTitle, setEditingTitle] = useState(item.title);
+  const [isEditing, setIsEditing] = useState(false);
+  const focusRef = useRef<HTMLInputElement>(null);
+
+  const handleChangeItem = (itemToChange:
+  { [key: string]: boolean | string }) => {
+    setIsProcessing([...isProcessing, id]);
+    setTodos(todos.map(todo => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          ...itemToChange,
+        };
+      }
+
+      return todo;
+    }));
+  };
+
+  const editTitle = (titleToEdit: string = editingTitle) => {
+    if (!editingTitle) {
+      setTodos(todos.filter(todo => todo.id !== id));
+    } else {
+      handleChangeItem({ title: titleToEdit });
+    }
+
+    setIsEditing(false);
+  };
+
+  const handleEditTitle = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      editTitle();
+    }
+
+    if (e.key === 'Escape') {
+      editTitle(title);
+    }
+  };
+
+  const getClassName = () => {
```

Comment:
Instead of mutating the className variable, consider using the ternary operator for assignment.

Reply:
Using a ternary operator on 3 options looks very confusing